curriculum/challenges/english/25-front-end-development/lab-reusable-footer/673b02b03134b04637bf7055.md

```diff
@@ -0,0 +1,274 @@
+---
+id: 673b02b03134b04637bf7055
+title: Build a Reusable Footer
+challengeType: 14
+dashedName: build-a-reusable-footer
+demoType: onClick
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should export a `Footer`component.
+2. Your `Footer` component should return a `footer` element that contains all the other elements.
+3. Your `footer` element should have:
+
+- At least three unordered lists, each with at least two list items.
+- At least one paragraph element with a copyright `©` symbol.
+- At least three links with the `href` value set to `#` and the link content set to an icon or text of your choice.
+
+# --hints--
+
+You should export a `Footer` component.
+
+```js
+const mockedComponent = Enzyme.mount(React.createElement(window.index.Footer));
```

Comment:
I'm not sure what the issues with using React 18 and the enzyme-adapter-16 are going to be, but it's very likely that there will be some.

If we're lucky, it will be obvious when we try and write tests, but I don't know enough about how React 18 and Enzyme are incompatible to say.

I don't have an alternative right now, but we may find that we need one.

Reply:
I think it will work for these first few react sections. I didn't look into it much, but I think we should be able to replace enzyme with react testing library. Or, probably add it along side to start.

Reply:
It's tricky because RTL is designed to discourage the kind of testing we want to do. Namely, digging into the implementation details.

Don't worry about it for now, though. It's just something to keep in mind and not be too surprised about if it misbehaves.